mlx_graphs/nn/models/node2vec.py

```diff
@@ -0,0 +1,165 @@
+import mlx.core as mx
+import mlx.nn as nn
+
+from mlx_graphs.utils.sorting import sort_edge_index
+
+try:
+    from mlx_cluster import random_walk, rejection_sampling
+except ImportError:
+    raise ImportError(
+        "mlx_cluster is required for performing random walks",
+        "run `pip install mlx_cluster`",
+    )
+import numpy as np
+
+
+class Node2Vec(nn.Module):
+    r"""The Node2Vec model from the
+    `"node2vec: Scalable Feature Learning for Networks"
+    <https://arxiv.org/abs/1607.00653>`_ paper where random walks of
+    length :obj:`walk_length` are sampled in a given graph, and node embeddings
+    are learned via negative sampling optimization
+
+    Args:
+        edge_index (mx.array): The edge indices.
+        embedding_dim (int): The size of each embedding vector.
+        walk_length (int): The walk length.
+        context_size (int): The actual context size which is considered for
+            positive samples. This parameter increases the effective sampling
+            rate by reusing samples across different source nodes.
+        num_nodes (int): Number of nodes in a graph
+        walks_per_node (int, optional): The number of walks to sample for each
+            node. (default: :obj:`1`)
+        p (float, optional): Likelihood of immediately revisiting a node in the
+            walk. (default: :obj:`1`)
+        q (float, optional): Control parameter to interpolate between
+            breadth-first strategy and depth-first strategy (default: :obj:`1`)
+        num_negative_samples (int, optional): The number of negative samples to
```

Comment:
you can safely remove the types from here as they're inferred from the __init__ signature